### PR TITLE
fix: accept seed.length == 29, slice returned bytes, implement bytesToMnemonic()

### PR DESCRIPTION
**Pull request by DaWe35**

### CODE REVIEW

File: lib/mnemonics.js [hunk 1]
```diff
@@ -27,16 +28,71 @@ function intToBytes (i) {
 }
 
 function mnemonicToBytes (mnemonic) {
-    if (mnemonic.split(' ').length !== 28) return 'Invalid mnemonic length';
+    if (mnemonic.split(' ').length !== 28 && mnemonic.split(' ').length !== 29) return 'Invalid mnemonic length';
```
Comment (wkibbler) [inline lib/mnemonics.js:31]: what's this for?

File: lib/mnemonics.js [hunk 2]
```diff
@@ -27,16 +28,71 @@ function intToBytes (i) {
...
 
 }
 
+
+
+// SeedToString converts a wallet seed to a human friendly string. (StringToSeed() in wallet.go)
+function bytesToMnemonic (bytes) {
+    // console.log(typeof bytes)
+    var fullChecksum = blake2b(32).update(bytes).digest()
+    console.log('fullChecksum', "[" + fullChecksum.join(' ') + "]")
```
Comment (wkibbler) [inline lib/mnemonics.js:47]: Remove console.log

File: lib/mnemonics.js [hunk 3]
```diff
@@ -27,16 +28,71 @@ function intToBytes (i) {
...
+    var checksumSeed = Uint8Array.from([...bytes, ...fullChecksum.slice(0,6)])
+    console.log('checksumSeed', "[" + checksumSeed.join(' ') + "]")
```
Comment (wkibbler) [inline lib/mnemonics.js:49]: Remove console.log

File: lib/mnemonics.js [hunk 4]
```diff
@@ -27,16 +28,71 @@ function intToBytes (i) {
...
+	var phrase = toPhrase(checksumSeed)
+	return phrase.join(' ')
+}
+
+function toPhrase (entropy) {
+	if (entropy.length == 0) {
+		return false
+	}
+    var intEntropy = bytesToInt(entropy)
+    console.log("intEntropy ", intEntropy.toFixed())
```
Comment (wkibbler) [inline lib/mnemonics.js:59]: Remove console.log

File: lib/KeyPair.js [hunk 5]
```diff
@@ -54,7 +54,7 @@ function generateFromSeed (seed) {
   return deriveAddress(Buffer.from(keypair.secretKey).toString('hex'))
 }
 
-// derives an address from the given private key and return an object with address, privateKey, and publicKey
+// generates keypairs from nemonic. First index: [0,0,0,0,0,0,0,0]
```
Comment (wkibbler) [inline lib/KeyPair.js:57]: spelling: nemonic -> mnemonic

**Review by wkibbler — CHANGES_REQUESTED** [review]

File: lib/mnemonics.js (hunk 1 above)
Comment (DaWe35) [reply lib/mnemonics.js:31]: 28 and 29-word seeds are valid seeds in Sia. Source: https://gitlab.com/NebulousLabs/Sia/-/merge_requests/3249/diffs#65ca216301490a8eb1d019e6389b24b1a1310dc4_496_513 (SiaCentral wallet also generates 29-word seeds)

**Pull request merged**